robusta_krr/core/runner.py

```diff
@@ -65,12 +65,35 @@ def _get_prometheus_loader(self, cluster: Optional[str]) -> Optional[PrometheusM
 
         return result
 
-    def _greet(self) -> None:
+    @staticmethod
+    def __parse_version_string(version: str) -> tuple[int, ...]:
+        version_trimmed = version.replace("-dev", "").replace("v", "")
+        return tuple(map(int, version_trimmed.split(".")))
+
+    def __check_newer_version_available(self, current_version: str, latest_version: str) -> bool:
+        try:
+            current_version_parsed = self.__parse_version_string(current_version)
+            latest_version_parsed = self.__parse_version_string(latest_version)
+
+            if current_version_parsed < latest_version_parsed:
+                return True
+        except Exception:
+            logger.exception("An error occurred while checking for a new version")
```

Comment:
is an exception thrown if latest version is None?
If we are running in a cluster the IP might be blocked and this could always be printed
Will the logs from this be an issue with how runner formats the krr results?

Reply:
I guess this isnt run since we run it with -q, but I just want to check

Reply:
You are actually correct, better write that as a debug message maybe

Reply:
No, actually we have a try catch in the function that gets the version
If it fails, it returns None

So this exception log (also not critical) is just because we do not expect to get a string with some invalid format here.

Still, I will make that a debug with a traceback, as someone at somepoint might set the invalid release version